### ISSUE TITLE
Doesn't compile in win-builder

### ISSUE THREAD

**Issue by Lobz** [Lobz/facilitation#55]
Warnings are all of this kind:
```
warning: delegating constructors only available with -std=c++11 or -std=gnu++11
```
Error is:
```
Rcpp/internal/wrap.h:526:26: error: invalid conversion from 'long long int' to 'SEXP' [-fpermissive]
```
Trying to figure out why this happens. Similar issues:
https://stackoverflow.com/questions/23395005/rcpp-long-long-unsigned-int-error-issue
Trying to follow this guide right now:
https://cran.r-project.org/doc/manuals/r-release/R-admin.html#Customizing-package-compilation

**Comment by Lobz** [comment]
```
d:/Compiler/gcc-4.9.3/mingw_64/bin/g++ -m64 -I"D:/RCompile/recent/R/include" -DNDEBUG  -I"d:/RCompile/CRANpkg/lib/3.5/Rcpp/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -pedantic -O2 -Wall  -mtune=core2 -c main.cpp -o main.o
In file included from d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/RcppCommon.h:195:0,
                 from d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp.h:27,
                 from Arena.h:9,
                 from Species.h:4,
                 from Individual.h:4,
                 from main.cpp:1:
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/internal/wrap.h: In instantiation of 'SEXPREC* Rcpp::internal::wrap_dispatch_unknown_iterable(const T&, Rcpp::traits::false_type) [with T = long long int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]':
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/internal/wrap.h:732:100:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch_unknown(const T&, Rcpp::traits::false_type) [with T = long long int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]'
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/internal/wrap.h:770:105:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch_eigen(const T&, Rcpp::traits::false_type) [with T = long long int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]'
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/internal/wrap.h:787:89:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch_unknown_importable(const T&, Rcpp::traits::false_type) [with T = long long int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]'
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/internal/wrap.h:807:108:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch(const T&, Rcpp::traits::wrap_type_unknown_tag) [with T = long long int; SEXP = SEXPREC*]'
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/internal/wrap_end.h:30:110:   required from 'SEXPREC* Rcpp::wrap(const T&) [with T = long long int; SEXP = SEXPREC*]'
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/api/meat/DataFrame.h:26:36:   required from 'Rcpp::DataFrame_Impl<StoragePolicy>::DataFrame_Impl(const T&) [with T = long long int; StoragePolicy = Rcpp::PreserveStorage]'
main.cpp:58:27:   required from here
d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/internal/wrap.h:526:26: error: invalid conversion from 'long long int' to 'SEXP' [-fpermissive]
                 SEXP x = object;
```

**Comment by Lobz** [comment]
main.cpp:58:
```
52     if(restore){
 53         populated = arena->populate(history);
 54     }
 55     else{
 56         populated = arena->populate(in);
 57     }
 58     if(!populated) return NULL;
 59 
 60     while(arena->getTotalTime() < maxtime && test){
 61         test = arena->turn();
 62         numturns++;
 63         if(arena->getTotalAbundance() > maxpop) {
 64             Rcpp::warning("Maximum population reached. Stopping...");
 65             test = false;
 66         }
 67     }
 68 
 69     ret = arena->finalStatus();
 70 
 71     return Rcpp::DataFrame::create(Rcpp::Named("sp")=ret->sp_list,Rcpp::Named("id")=ret->id_l    ist,
 72             Rcpp::Named("x")=ret->x_list,Rcpp::Named("y")=ret->y_list,
```

**Comment by Lobz** [comment]
I asked a [question in StackOverflow](https://stackoverflow.com/questions/48668357/how-to-make-sure-win-builder-builds-my-package-with-c11) about the first issue. Answers pointed to use of `Makevars.win` or to adding a line `SystemRequirements: C++11` in `DESCRIPTION`, but both were not very hopeful that I would manage to solve the problem with compiler options.

**Comment by Lobz** [comment]
Solved in v0.5.1 (by commit 8f0dc31a45d64fd59c7f2173a14e2b84c0cb0233 )
